rudder-web/src/main/webapp/javascript/rudder/angular/nodeProperties.js

```diff
@@ -0,0 +1,158 @@
+/*
+*************************************************************************************
+* Copyright 2017 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+* 
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+* 
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+* 
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+* 
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+var app = angular.module('nodeProperties', ['DataTables', 'monospaced.elastic']);
+
+app.controller('nodePropertiesCtrl', function ($scope, $http, $compile) {
+  //Initialize scope
+  $scope.properties;
+  $scope.nodeId;
+  $scope.idTable          = "#nodePropertiesTab";
+  $scope.urlAPI           = contextPath + '/secure/api/latest/nodes/';
+  $scope.newProperty      = {'name':"", 'value':""}
+  $scope.selectedProperty = {'name':"", 'index':""};
+  $scope.alreadyUsed      = false;
+  $scope.errorSaving      = false;
+  $scope.errorDeleting    = false;
+
+  $scope.resetNewProperty = function(){
+    $scope.newProperty = {'name':"", 'value':""};
+  }
+
+  var columnDefs = [
+    {"aTargets":[0], "mDataProp": "name" , "sWidth": "250px", "sTitle" : "Name"  }
+  , {
+    "aTargets":[1],
+    "mDataProp": "value",
+    "sTitle" :"Value",
+    "fnCreatedCell" : function (nTd, sData, oData, iRow, iCol) {
+      var pre = $("<pre class='json-beautify' onclick='$(this).toggleClass(\"toggle\")'></pre>");
+      var content = sData != null && typeof sData === 'object' ? JSON.stringify(sData) : sData;
+      $(nTd).empty();
+      pre.html(content)
+      $(nTd).prepend(pre);
+     }
+   }
+  , {
+    "aTargets":[2],
```

Comment:
Sad about the third column (you can see it in the table header )

Maybe one day we will not use anymore the datatable directive`




```html
 <table><thead>...</thead><tbody><tr ng-repeat="property in properties">...<tr></tbody></table)
```